### ISSUE TITLE
Add force sync option to sync button on the application detail page

### ISSUE THREAD

**Issue by cakecatz** [pipe-cd/pipe#1159]
Before we start this issue, we need API to `force sync`.

**What would you like to be added**:

**Why is this needed**:

**Comment by cakecatz** [comment]
Ref #128

**Comment by cakecatz** [comment]
@nghialv What label should we use for the sync button?

I found SyncStrategy enum, that has `Auto/Quick/Pipeline`.
So should we use that enum as the button label like this?

![image](https://user-images.githubusercontent.com/6136383/102573441-d48ac800-4131-11eb-97d8-ea7374fc0f65.png)


If you have any suggestions, please give me some advice.

**Comment by nghialv** [comment]
The default value is `Sync` and 2 options are `Quick Sync` and `Sync with Pipeline`.

`Sync with Pipeline` is more accurate compared to `Pipeline Sync`.
But `Pipeline Sync` looks shorter and simpler. I have a feeling `Pipeline Sync` and `Quick Sync`  is a perfect pair.

@nakabonne @khanhtc1202 What are your thoughts?

**Comment by khanhtc1202** [comment]
same as to mention above, I think `Pipeline Sync` is a better name than `Sync with Pipeline`. But I have another issue, `Auto Sync` and `Pipeline Sync` may confused users since it's ambiguous between them.

**Comment by nghialv** [comment]
About the button label, I think "Sync" is better than "Auto Sync".
So we should keep the current UI. "SYNC" will be shown as the default, and the user can click on the pull-down icon to select other options: `SYNC`, `QUICK SYNC` or `PIPELINE SYNC`.
